allennlp/tests/modules/contextual_encoder/contextual_encoder_test.py

```diff
@@ -0,0 +1,77 @@
+# pylint: disable=invalid-name,no-self-use
+import torch
+import numpy as np
+
+from allennlp.common.testing import AllenNlpTestCase
+from allennlp.modules.contextual_encoder.character_encoder import CharacterEncoder
+from allennlp.modules.contextual_encoder.contextual_encoder import (
+        CharLevelContextualEncoder, TokenLevelContextualEncoder)
+from allennlp.modules.seq2seq_encoders import PytorchSeq2SeqWrapper
+from allennlp.modules.token_embedders import Embedding
+
+class TestContextualEncoder(AllenNlpTestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.character_encoder = CharacterEncoder(
+                activation='relu',
+                embedding_dim=4,
+                filters=[[1, 4], [2, 8], [3, 16], [4, 32], [5, 64]],
+                max_characters_per_token=50,
+                num_characters=262,
+                num_highway=2,
+                projection_dim=16,
+                projection_location='after_cnn'
+        )
+
+        lstm = torch.nn.LSTM(bidirectional=True,
+                             num_layers=3,
+                             input_size=16,
+                             hidden_size=10,
+                             batch_first=True)
+        self.seq2seq = PytorchSeq2SeqWrapper(lstm)
+
+
+    def test_char_level_contextual_encoder(self):
+        cew = CharLevelContextualEncoder(
+                encoder=self.seq2seq,
+                character_encoder=self.character_encoder,
+                num_layers=3,
```

Comment:
I think `num_layers` should be 4.. it isn't a problem here, since it's never tested.
The `num_layers` thing is so that the contextual encoder can be automatically wrapped with the `Elmo` scalar weighting to make a token embedder.